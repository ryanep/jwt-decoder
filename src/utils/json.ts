const JSON_INDENTATION = 2;

export const parseJson = <TData>(jsonString: string): TData | null => {
  try {
    return JSON.parse(jsonString) as TData;
  } catch {
    return null;
  }
};

export const formatJson = (json: any): string => {
  return json ? JSON.stringify(json, null, JSON_INDENTATION) : null;
};
