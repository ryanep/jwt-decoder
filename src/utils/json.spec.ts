import { parseJson, formatJson } from './json';

describe('utils/json', () => {
  describe('parseJson', () => {
    it('should return null with an invalid JSON payload', () => {
      const jsonString = '__INVALID_JSON__';
      const result = parseJson(jsonString);
      expect(result).toBe(null);
    });

    it('should return object with a valid JSON payload', () => {
      const jsonString = '{ "test": true }';
      const result = parseJson(jsonString);
      expect(result).toMatchObject({
        test: true,
      });
    });
  });

  describe('formatJson', () => {
    it('should return null is input is null', () => {
      const json: any = null;
      const result = formatJson(json);
      expect(result).toBe(null);
    });

    it('should return formatted JSON string with valid object', () => {
      const json = {
        test: true,
      };
      const result = formatJson(json);
      expect(result).toMatchSnapshot();
    });
  });
});
