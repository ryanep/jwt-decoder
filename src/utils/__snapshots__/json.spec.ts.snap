// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/json formatJson should return formatted JSON string with valid object 1`] = `
"{
  \\"test\\": true
}"
`;
